### PR TITLE
Implementation of transaction builder package

### PR DESCRIPTION
**Pull request by pablocampogo**
Implements transaction builder package with `Submit` function that creates the transaction from raw input and `CreateTransaction` function that returns parsed input for creating a transaction.

Example of usage:

```
wallet, err := signer.NewRandomWallet()
if err != nil {
	return err
}

txBuilder := transactionbuilder.NewPocketTransactionBuilder(provider.NewJSONRPCProvider("https://dummy.com", []string{"https://dummy.com"}), wallet)

// This can be any of the messages implemented on transaction-builder/models.go
msgSend, err := NewMsgSend("b50a6e20d3733fb89631ae32385b3c85c533c560", "b50a6e20d3733fb89631ae32385b3c85c533c561", 21)
if err != nil {
	return err
}

output, err = txBuilder.Submit(transactionbuilder.Mainnet, msgSend, nil)
if err != nil {
	return err
}

DoStuff(output)
```

Closes #37

### CODE REVIEW

**Review by crisog — APPROVED** [review]
Looks good! I used it and sent a few transactions on my localnet and they worked fine.

File: pkg/transaction-builder/pocket_transaction_builder.go [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+	return &PocketTransactionBuilder{
+		provider: provider,
+		signer:   signer,
+	}
+}
+
+func getOptionalParams(options *TransactionOptions) (string, string, int64) {
+	memo := ""
+	coinDenom := Upokt
+	fee := int64(10000)
```
Comment (rem1niscence) [inline pkg/transaction-builder/pocket_transaction_builder.go:47]: what that number means?

Comment (Evalir) [reply pkg/transaction-builder/pocket_transaction_builder.go:47]: Seems like it's the chain base fee. Should prob be a constant up above.

File: pkg/transaction-builder/models.go [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+	"encoding/hex"
+
+	"github.com/pokt-network/pocket-core/crypto"
+	coreTypes "github.com/pokt-network/pocket-core/types"
+	appsType "github.com/pokt-network/pocket-core/x/apps/types"
+	nodesTypes "github.com/pokt-network/pocket-core/x/nodes/types"
+)
+
+// NewMsgSend returns message for send transaction
+func NewMsgSend(fromAddress, toAddress string, amount int64) (TxMsg, error) {
```
Comment (Evalir) [inline pkg/transaction-builder/models.go:13]: A bit of a bikeshed here, but if we're not including `Msg` in all of the other methods, we shouldn't include it here either. 
```suggestion
func NewSend(fromAddress, toAddress string, amount int64) (TxMsg, error) {
```

File: pkg/transaction-builder/pocket_transaction_builder_test.go [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+
+	httpmock.Activate()
+	defer httpmock.DeactivateAndReset()
+
+	wallet, err := signer.NewRandomWallet()
+	c.NoError(err)
+
+	txBuilder := NewPocketTransactionBuilder(provider.NewJSONRPCProvider("https://dummy.com", []string{"https://dummy.com"}), wallet)
+
+	msgSend, err := NewMsgSend("b50a6e20d3733fb89631ae32385b3c85c533c560", "b50a6e20d3733fb89631ae32385b3c85c533c561", 21)
```
Comment (Evalir) [inline pkg/transaction-builder/pocket_transaction_builder_test.go:69]:
```suggestion
	msgSend, err := NewSend("b50a6e20d3733fb89631ae32385b3c85c533c560", "b50a6e20d3733fb89631ae32385b3c85c533c561", 21)
```

**Review by Evalir — APPROVED** [review]
Looks pretty solid IMO!

File: pkg/transaction-builder/pocket_transaction_builder.go (hunk 1 above)
Comment (pablocampogo) [reply pkg/transaction-builder/pocket_transaction_builder.go:47]: I'll add a comment, it's the default fee for transactions

File: pkg/transaction-builder/enums.go [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+
+// CoinDenom enum that represents all coin denominations of Pocket
+type CoinDenom string
+
+const (
+	// Upokt represents upokt denomination
+	Upokt CoinDenom = "upokt"
+	// Pokt represents pokt denomination
+	Pokt CoinDenom = "pokt"
+)
```
Comment (Evalir) [inline pkg/transaction-builder/enums.go:11]: Was wondering: wdyt about adding a `ChainID` enum here as well? It's pretty extra but would be handy as it's not really documented anywhere what `ChainID` corresponds to.

File: pkg/transaction-builder/pocket_transaction_builder.go [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+	}
+
+	return &provider.SendTransactionInput{
+		Address:     t.signer.GetKeyManager().GetAddress(),
+		RawHexBytes: signedTX,
+	}, nil
+}
+
+// Submit does the transaction from raw input
+func (t *PocketTransactionBuilder) Submit(chainID string, txMsg TxMsg, options *TransactionOptions) (*provider.SendTransactionOutput, error) {
```
Comment (Evalir) [inline pkg/transaction-builder/pocket_transaction_builder.go:147]: Realized that we should probably include sane defaults for `chainID` and some of the transaction options:
- for the `fee` you probably wanna always pay the base fee in uPOKT, 10k.
- For chainID, we should probably default to mainnet and have an enum handy (see above).

Comment (pablocampogo) [reply pkg/transaction-builder/pocket_transaction_builder.go:147]: All optional params have default values setted on `getOptionalParams`

File: pkg/transaction-builder/enums.go (hunk 4 above)
Comment (pablocampogo) [reply pkg/transaction-builder/enums.go:11]: I was told by @rem1niscence that these values are not static over time, if they don't I find an enum pretty useful too

File: pkg/transaction-builder/pocket_transaction_builder.go (hunk 5 above)
Comment (Evalir) [reply pkg/transaction-builder/pocket_transaction_builder.go:147]: Ah cool!

File: pkg/transaction-builder/enums.go (hunk 4 above)
Comment (Evalir) [reply pkg/transaction-builder/enums.go:11]: What do you mean they're not static over time?

**Review by Evalir — COMMENTED** [review]
Before merging, let's also make sure to run testnet/mainnet test transactions to make sure everything is OK. @crisog and I can help with that

File: pkg/transaction-builder/enums.go (hunk 4 above)
Comment (crisog) [reply pkg/transaction-builder/enums.go:11]: `ChainID` is set at the genesis file, and could be any string (see [here](https://github.com/pokt-network/pocket-network-genesis/blob/master/mainnet/genesis.json#L3)). The enum could be useful to give a general sense of what is expected, assuming people don't go straight to docs.

Comment (crisog) [reply pkg/transaction-builder/enums.go:11]: As long as you don't lock it to the enum, it's fine. You should still be allowed to use any string, but 95% people are using the enum (either connecting to mainnet, testnet or a localnet)

**Pull request merged**